gsoc2018-bharat/src/evaluate.py

```diff
@@ -0,0 +1,205 @@
+    description: Description that is to be encoded
+    """
+    global wv, table
+    d = description.translate(table).lower().split()
+    r = np.array(list(map(lambda x: wv.get_vector(x), d)),
+                 dtype=np.float32)
+    return r.mean(axis=0)
+
+
+def distance_encoder(description):
+    """
+    Encodes the description using simple vector averaging
+
+    Argument
+    --------
+    description: Description that is to be encoded
+    """
+    global wv, table
+    d = description.translate(table).lower().split()
+    r = list(map(lambda x: wv.get_vector(x), d))
+    r = np.array([idx / (i + 1) for i, idx in enumerate(r)],
+                 dtype=np.float32)
+    return r.mean(axis=0)
+
+
+def title_mean(label):
+    """
+    Generates the embedding by averaging the vectors
+    representing the words present in the title
+
+    Argument
+    --------
+    title: Title/Label of the entity
+    """
+    global wv, table
+    label = label.translate(table).lower().replace('_', ' ').split()
+    r = np.array(list(map(lambda x: wv.get_vector(x), label)),
+                 dtype=np.float32)
+    return r.mean(axis=0)
+
+
+def abstract_encoder(label):
+    """
+    Encodes the abstract using the RNN network
+
+    Argument
+    --------
+    dictionary: Dictionary containing all abstracts
+    label: Label for the entity that is to be encoded
+    """
+    global dictionary, wv, table
+    model = torch.load('model/description_encoder')
+    # label = label.lower()
+    try:
+        abstract = dictionary[label]
+        d = abstract.translate(table).lower()
+        d = d.replace('resource/', '').split()
+        r = np.array(list(map(lambda x: wv.get_vector(x), d)),
+                     dtype=np.float32)
+        hidden = model.init_hidden()
+    except KeyError:
+        return np.random.randn(100)
+    try:
+        for word in r:
+            p, hidden = model(Variable(torch.tensor([[word]])),
+                              hidden)
+            p = p[0][0].detach().numpy()
+        return p
+    except (KeyError, IndexError, TypeError) as _:
+        return np.random.randn(100)
+
+
+def evaluate():
+    global dictionary, wv
+    count = 0
+    scores = np.zeros(7)
+    cos_scores = np.zeros(7)
+    itr = len(dictionary)
+    logging.info('running evaluation for {0} samples'.format(itr))
+    for key in dictionary:
+        progress = (count / itr) * 100
+        print('INFO : evaluated {0:.1f} %'.format(progress), end='\r')
+        d = dictionary[key].split('resource/')
+        d = [idx.split()[0].translate(table).lower() for idx in d[1:]]
+        try:
+            r = np.array(list(map(lambda x: wv.get_vector(x), d)),
+                         dtype=np.float32)
+        except KeyError:
+            itr -= 1
+            continue
+        if np.any(np.isnan(r)):
+            itr -= 1
+            continue
+        else:
+            if r.ndim == 2:
+                try:
+                    r = r.mean(axis=0).reshape(1, -1)
```

Comment:
This whole function as it stands is very hard to understand. Probably you can isolate this whole code block into a function and document what each index means. Also use way more descriptive names for your varirables, e.g. `random_vec` instead of `vec5`, `zero_vec` instead of `vec6`